### ISSUE TITLE
Running Task in self-managed agent behind a Proxy 

### ISSUE THREAD

**Issue by omaraladin** [shayki5/azure-devops-create-pr-task#153]
Hi,

Description: 
when running the task in a self-managed agent, it throws an error:
`##[error]The remote name could not be resolved: 'dev.azure.com'`
the domain is bypassed through my proxy

How to reproduce:
Include the task in a pipeline and specify demands to make the pipeline run in a self-managed agent that is running behind a proxy

Possible fixes:
Include an optional proxy-Parameter in the task, that if added proxy will be loaded in the task runtime

**Comment by shayki5** [comment]
Hi @omaraladin, is it similar to #72?
If yes, can you provide me which details need to add? (I just not familiar with proxy...)

**Comment by omaraladin** [comment]
> Hi @omaraladin, is it similar to #72? If yes, can you provide me which details need to add? (I just not familiar with proxy...)

Yeah it is similar to 72

you need to load the proxy details, in the code at runtime, because in proxy-environments if proxy is not loaded, **Internet URLs** will not be reached from the **Request origin** (the self-managed agent running your PRCreate task in my case)

how I think you could resolve it, as PS is used to code the task:
- Add  optional proxyParameter in the task, and if added, Load into **defaultproxy environment variable** on the Task Powershell runtime:
`$HTTP_PROXY="http://proxy.company.com:8080"`
`$HTTPS_PROXY="http://proxy.company.com:8080"`
Ref:
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-webrequest?view=powershell-7.3#notes

- Add  optional proxyParameter in the task, and if added, load it as proxy-URL on your underlying PS Commands runtime:
example:
 `Invoke-WebRequest https://dev.azure.com -Proxy "http://proxy.company.com:8080" -ProxyCredential $ProxyCreds`
Ref:
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-webrequest?view=powershell-7.3#-proxy

**Comment by omaraladin** [comment]
Just a reminder, availing this option, may open the path for the extension to be used by Highly Regulated Eenvironments organizations, which will be valuable to both the Extension and these Organizations

**Comment by shayki5** [comment]
@omaraladin I using `Invoke-RestMethod` so I think `-Proxy` and `-ProxyCredential` parameters will do the job, no?

See here: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod?view=powershell-7.3

**Comment by omaraladin** [comment]
yeah, should do the job, but make sure to add this flag **only** when the user specifies that a proxy is actually being used.

**Comment by spaasis** [comment]
Any progress or workarounds for this?

**Comment by shayki5** [comment]
@omaraladin @spaasis can you try to run a simple Azure DevOps Rest API with Powershell and add the proxy param, for check if it will work?
